packages/salesforcedx-vscode-apex-replay-debugger/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,62 @@
+import vscode = require('vscode');
+import TelemetryReporter from 'vscode-extension-telemetry';
+
+export class TelemetryService {
+  private static instance: TelemetryService;
+  private context: vscode.ExtensionContext | undefined;
+  private reporter: TelemetryReporter | undefined;
+  private isTelemetryEnabled: boolean;
+
+  constructor() {
+    this.isTelemetryEnabled = false;
+  }
+
+  public static getInstance() {
+    if (!TelemetryService.instance) {
+      TelemetryService.instance = new TelemetryService();
+    }
+    return TelemetryService.instance;
+  }
+
+  public initializeService(
+    context: vscode.ExtensionContext,
+    isTelemetryEnabled: boolean
+  ): TelemetryReporter | undefined {
+    this.context = context;
+    this.isTelemetryEnabled = isTelemetryEnabled;
+
+    // TelemetryReporter is not initialized if user has disabled telemetry setting.
+    if (this.reporter === undefined && this.isTelemetryEnabled) {
+      const extensionPackage = require(this.context.asAbsolutePath(
```

Comment:
Is this to get the ai key? Or what else does it need from package.json?

Reply:
NVM. I see that you are using the `require` technique to read this JSON into an object to be used later.